pyseir/inference/model_fitter.py

```diff
@@ -619,38 +654,33 @@ def fit(self):
         self.mle_model = self.run_model(**{k: self.fit_results[k] for k in self.model_fit_keys})
 
     @classmethod
-    def run_for_fips(cls, fips, n_retries=3, with_age_structure=False):
+    def run_for_region(cls, regional_input: RegionalInput, n_retries=3):
         """
-        Run the model fitter for a state or county fips code.
+        Run the model fitter for a regional_input.
 
         Parameters
         ----------
-        fips: str
-            2-digit state or 5-digit county fips code.
+        region: pipeline.Region
         n_retries: int
             The model fitter is stochastic in nature and a seed cannot be set.
             This is a bandaid until more sophisticated retries can be
             implemented.
-        with_age_structure: bool
-            If True run model with age structure.
 
         Returns
         -------
         : ModelFitter
         """
         # Assert that there are some cases for counties
-        if len(fips) == 5:
-            _, observed_new_cases, _ = load_data.load_new_case_data_by_fips(
-                fips, t0=datetime.today()
-            )
+        if regional_input.region.is_county():
+            _, observed_new_cases, _ = regional_input.load_new_case_data(t0=datetime.today())
             if observed_new_cases.sum() < 1:
                 return None
 
         try:
             retries_left = n_retries
             model_is_empty = True
             while retries_left > 0 and model_is_empty:
-                model_fitter = cls(fips=fips, with_age_structure=with_age_structure)
+                model_fitter = cls(regional_input=regional_input)
```

Comment:
fwiw woudl prefer to call without `regional_input` since its not a kwarg

Reply:
Done.